src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1607,10 +1606,31 @@ public async Task InboundWindowSize_Exceeded_Throw()
                 async uri =>
                 {
                     using HttpClient client = CreateHttpClient();
-                    using HttpResponseMessage response = await client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead);
 
-                    // Keep client open until server is done.
-                    await semaphore.WaitAsync(10000);
+                    Exception requestException = await Assert.ThrowsAsync<HttpRequestException>(async () =>
+                    {
+                        // GetAsync() tends to throw when running under high load.
```

Comment:
This sounds like we're writing this test to workaround another bug with GetAsync().  Is there an issue about this problem with GetAsync().

Reply:
The behavior is inconsistent due to a race between the background task that reads frames and GetAsync reading headers. I don't have any strong opinions on if this inconsistency should be considered a bug or not.

If the background task happens to read enough to exceed the receive window before GetAsync starts reading headers, it will throw and GetAsync will bubble it up. Otherwise, reading `Content` will bubble it up.

Reply:
>The behavior is inconsistent due to a race between the background task that reads frames and GetAsync reading headers. 

The sounds like a product bug to me.  @wfurt is there an issue for this?  If not, please open one up. Then reference that issue here in this test.

Modifying a brittle test (like this one) temporarily to work around a product bug is ok for now as long as we reference the product bug we need to fix. In which case, we'll be able to go back and fix up the test to remove the workaround.

Reply:
I'm not clear on what the bug is. Are you saying this is an exceptional situation, and the exception is either emerging from GetAsync's task vs ReadAsStringAsync's task, depending on how quickly the client receives the faulty content?

Reply:
@stephentoub correct. One of them needs to throw. I guess the question can be stated as:

If the receive window has been exceeded, BUT all the headers can be read within the desired receive window, should `GetAsync` throw immediately or should it hold that exception until we read `Content`?

My (weak) opinion is that it's better to throw sooner than later, but I can imagine scenarios where a user might prefer to be able to at least read the headers.

Reply:
@davidsh any thoughts on ways we might make the test less brittle?

Reply:
> should GetAsync throw immediately or should it hold that exception until we read Content?

Thanks. My preference is consistency over expediency, so I'd prefer we not throw until the relevant content is processed. However, at this point u wouldn't complicate the implementation significantly to achieve that. While it's arguably a bug, it's a minor one, and one that's unlikely to actually affect the majority of consumption.

Reply:
>While it's arguably a bug, it's a minor one, and one that's unlikely to actually affect the majority of consumption.

We should still open an issue if there is a product bug here.  We might decide not to fix it for .NET Core 3.0 release, but we should track it.

Reply:
@scalablecory 

>@davidsh any thoughts on ways we might make the test less brittle?

I think we should first isolate what is a product bug versus a possible test bug (i.e. brittle test).  Once we do that, then I'll be in a better position to suggest ways to make the test less brittle.

Reply:
Some recent commits on a different PR addressing HTTP2 test issues might be good to use here:
https://github.com/dotnet/corefx/commit/e6931edc34353f52aafdff1421f16de5e5921405

Reply:
Filed issue #38759 to track this.